### PR TITLE
fix zocl when iommu attach fails

### PR DESCRIPTION
**Pull request by larry9523**
This PR fixes two issues
1) When the IOMMU is present on amba bus, if we fail to attach iommu, we should clear flag in zocl. Otherwise, we will still work under IOMMU mode which is obviously wrong.
2) For any reason we fail to initialize iommu, do not fail the whole zocl probe but keep moving on after clearing all the resources allocated during iommu initialization.

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 1]
```diff
@@ -671,26 +700,12 @@ static int zocl_drm_platform_probe(struct platform_device *pdev)
...
-		end = geometry->aperture_end;
-
-		DRM_INFO("IOMMU aperture initialized (%#llx-%#llx)\n",
-				start, end);
+		/*
+		 * Note: we ignore the return value of zocl_iommu_init().
+		 * In the case of failing to initialize iommu, zocl
+		 * driver will keep working with iommu disabled.
+		 */
+		(void) zocl_iommu_init(zdev, pdev);
```
Comment (sonals) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: Should we at least warn if the IOMMU attach fails?

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 2]
```diff
@@ -497,6 +497,35 @@ static unsigned int zocl_poll(struct file *filp, poll_table *wait)
...
+static int zocl_iommu_init(struct drm_zocl_dev *zdev,
+		struct platform_device *pdev)
+{
+	struct iommu_domain_geometry *geometry;
+	u64 start, end;
+	int ret;
+
+	zdev->domain = iommu_domain_alloc(&platform_bus_type);
+	if (!zdev->domain)
+		return -ENOMEM;
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: For ENOMEM, why we still continue? I guess we should bail out right away?

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c (hunk 1 above)
Comment (larry9523) [reply src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: Yes we do log warning message if zocl_iommu_init() if attache fails

**Comment by larry9523** [comment]
retest this please

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c (hunk 2 above)
Comment (xdavidz) [reply src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: make sense. it is actually no-resource than no memory.

**Pull request merged**